phys2bids/phys2bids.py

```diff
@@ -116,163 +170,126 @@ def use_heuristic(heur_file, sub, ses, filename, outdir):
+
+    """
     options = _get_parser().parse_args(argv)
     # Check options to make them internally coherent
     # #!# This can probably be done while parsing?
-    # #!# Make filename check better somehow.
     options.indir = utils.check_input_dir(options.indir)
     options.outdir = utils.check_input_dir(options.outdir)
-    options.filename = utils.check_input_ext(options.filename, '.acq')
-    ftype = 'acq'
-    if not os.path.isfile(os.path.join(options.indir, options.filename)):
-        options.filename = utils.check_input_ext(options.filename[:-4], '.txt')
-        ftype = 'txt'
-
-    # #!# Change this to cases of and better message
-    print(f'File extension is .{ftype}')
+    options.filename, ftype = utils.check_input_type(options.filename,
+                                                     options.indir)
 
     if options.heur_file:
         options.heur_file = utils.check_input_ext(options.heur_file, '.py')
+        utils.check_file_exists(options.heur_file)
 
     infile = os.path.join(options.indir, options.filename)
-    outfile = os.path.join(options.outdir, os.path.basename(options.filename[:-4]))
-
     utils.check_file_exists(infile)
-    print('File exists')
+    outfile = os.path.join(options.outdir, os.path.basename(options.filename[:-4]))
 
-    # Read infos from file
+    # Read file!
     if ftype == 'acq':
-        from bioread import read_file
-
-        data = read_file(infile).channels
-        print_info_acq(options.filename, data)
+        from phys2bids.interfaces.acq import populate_phys_input
     elif ftype == 'txt':
-        header = print_info_txt(options.filename)
+        raise Exception('txt not yet supported')
+    else:
+        # #!# We should add a logger here.
+        raise Exception('Currently unsupported file type.')
+
+    print('Reading the file')
+    phys_in = populate_phys_input(infile, options.chtrig)
+    print('Reading infos')
+    utils.print_info(options.filename, phys_in)
+    # #!# Here the function viz.plot_channel should be called
+    # for the desired channels.
 
     # If file has to be processed, process it
     if not options.info:
+
+        # #!# Get option of no trigger! (which is wrong practice or Respiract)
+        phys_in.check_trigger_amount(options.thr, options.num_tps_expected,
+                                     options.tr)
+        print('Checking that the output folder exists')
+        utils.path_exists_or_make_it(options.outdir)
+        print('Plot trigger')
+        viz.plot_trigger(phys_in.timeseries[0], phys_in.timeseries[1],
+                         outfile, options)
+
+        # The next few lines remove the undesired channels from phys_in.
+        if options.chsel:
+            print('Dropping unselected channels')
+            for i in reversed(range(0, phys_in.ch_amout)):
+                if i not in options.chsel:
+                    phys_in.delete_at_index(i)
+
+        # If requested, change channel names.
+        if options.ch_name:
```

Comment:
I think this should be updated to:

```suggestion
        if options.ch_name is not None:
```

if you make the requested change in `phys2bids/cli/run.py`!